core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/StreamBatchSource.java

```diff
@@ -94,6 +94,10 @@
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     streamBatchConfig.validate();
     pipelineConfigurer.addStream(new Stream(streamBatchConfig.name));
+    // if no format is specified then default schema is used, if otherwise its based on input records.
+    if (streamBatchConfig.format == null) {
+      pipelineConfigurer.getStageConfigurer().setOutputSchema(DEFAULT_SCHEMA);
+    }
```

Comment:
yeah, have fixed it, the schema fields are added as separate fields and not as a single body field in transform, so doing the same.